### PR TITLE
Typing indicators support

### PR DESCRIPTION
**Pull request by lemieux**
This PR adds supports for typing indicators. 

anim

@dannytranlx @hebime @jugarrit @mspensieri

### CODE REVIEW

File: src/js/components/conversation.jsx [hunk 1]
```diff
@@ -135,8 +142,8 @@ export class ConversationComponent extends Component {
     };
 
     componentWillUpdate(nextProps) {
-        const {messages: currentMessages, isFetchingMoreMessages} = this.props;
-        const {messages: newMessages} = nextProps;
+        const {messages: currentMessages, isFetchingMoreMessages, typingIndicatorShown: currentTypingIndicatorShown} = this.props;
+        const {messages: newMessages, typingIndicatorShown:newTypingIndicatorShown} = nextProps;
```
Comment (mspensieri) [inline src/js/components/conversation.jsx]: Formatting here?

File: src/js/components/message.jsx [hunk 2]
```diff
@@ -86,10 +85,6 @@ export class MessageComponent extends Component {
             }
         }
 
-        if (hasQuickReplies) {
-            containerClass.push('sk-msg-has-quick-replies');
```
Comment (mspensieri) [inline src/js/components/message.jsx]: this class was unused?

File: src/js/reducers/app-state-reducer.js [hunk 3]
```diff
@@ -191,6 +195,32 @@ export function AppStateReducer(state = INITIAL_STATE, action) {
...
+                typingIndicatorShown: true,
+                typingIndicatorAvatarUrl: action.avatarUrl,
+                typingIndicatorName: action.name,
+                typingIndicatorTimeoutId: action.timeoutId
+            };
+        case AppStateActions.HIDE_TYPING_INDICATOR:
+            return {
+                ...state,
+                typingIndicatorShown: false,
+                typingIndicatorTimeoutId: null
```
Comment (mspensieri) [inline src/js/reducers/app-state-reducer.js]: should name and avatarUrl be reset here too?

File: src/js/components/conversation.jsx (hunk 1 above)
Comment (lemieux) [inline src/js/components/conversation.jsx]: ehh yeah but esformatter doesn't format that use case... I'll do it by hand

File: src/js/components/message.jsx (hunk 2 above)
Comment (lemieux) [inline src/js/components/message.jsx]: yeah it was doing nothing.

**Pull request merged**